syntax_and_semantics/operators.md

```diff
@@ -21,101 +53,467 @@ struct Vector2
+| `/` | division | `1 / 2` | yes |
+| `//` | floor division | `1 // 2` | yes |
+| `%` | modulus | `1 % 2` | yes |
+
+#### Additive
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `+` | addition | `1 + 2` | yes |
+| `&+` | wrapping addition | `1 &+ 2` | yes |
+| `-` | subtraction | `1 - 2` | yes |
+| `&-` | wrapping subtraction | `1 &- 2` | yes |
+
+### Other unary operators
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `!` | inversion | `!true` | no |
+| `~` | binary complement | `~1` | yes |
+
+### Shifts
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `<<` | shift left, append | `1 << 2`, `STDOUT << "foo"` | yes |
+| `>>` | shift right | `1 >> 2` | yes |
+
+### Binary
+
+<table>
+<thead>
+<tr>
+<th>Operator</th>
+<th>Description</th>
+<th>Example</th>
+<th>Overloadable</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td><code>&amp;</code></td>
+<td>binary AND</td>
+<td><code>1 &amp; 2</code></td>
+<td>yes</td>
+</tr>
+<tr>
+<td><code>|</code></td>
+<td>binary OR</td>
+<td><code>1 | 2</code></td>
+<td>yes</td>
+</tr>
+<tr>
+<td><code>^</code></td>
+<td>binary XOR</td>
+<td><code>1 ^ 2</code></td>
+<td>yes</td>
+</tr>
+</tbody>
+</table>
+
+### Equality
+
+Three base operators test equality:
+
+* `==`: Checks whether the values of the operands are equal
+* `=~`: Checks whether the value of the first operand matches the value of the
+second operand with pattern matching.
+* `===`: Checks whether the left hand operand matches the right hand operand in
+  [case equality](case.html). This operator is applied in `case ... when`
+  conditions.
+
+The first two operators also have inversion operators (`!=` and `!~`) whose
+semantical intention is just the inverse of the base operator: `a != b` is
+supposed to be equivalent to `!(a == b)` and `a !~ b` to `!(a =~ b)`.
+Nevertheless, these inversions can be defined with a custom implementation. This
+can be useful for example to improve performance (non-equality can often be
+proven faster than equality).
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `==` | equals | `1 == 2` | yes |
+| `!=` | not equals | `1 != 2` | yes |
+| `=~` | pattern match | `"foo" =~ /fo/` | yes |
+| `!~` | no pattern match | `"foo" !~ /fo/` | yes |
+| `===` | [case equality](case.html) | `/foo/ === "foo"` | yes |
+
+### Comparison
+
+| Operator | Description | Example | Overloadable |
+|---|---|---|---|
+| `<` | less | `1 < 2` | yes |
+| `<=` | less or equal | `1 <= 2` | yes |
+| `>` | greater | `1 > 2` | yes |
+| `>=` | greater or equal | `1 >= 2` | yes |
+| `<=>` | comparison | `1 <=> 2` | yes |
+
+### Logical
+
+<table>
+<thead>
+<tr>
+<th>Operator</th>
+<th>Description</th>
+<th>Example</th>
+<th>Overloadable</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td><code>&amp;&amp;</code></td>
+<td><a href="and.html">logical AND</a></td>
+<td><code>true &amp;&amp; false</code></td>
+<td>yes</td>
```

Comment:
I just noticed that it wrongly says that logical operators are overloadable. The compiler defines the typing and semantics of these operators.